test/parallel_api/algorithm/alg.sorting/sort_by_key_common.h

```diff
@@ -0,0 +1,250 @@
+    auto policy =  TestUtils::make_device_policy<Name>(q);
+    call_sort(policy, device_keys, device_vals, n, stability_tag, direction_tag);
+
+    // checking results
+    int host_out_keys[n] = {};
+    int host_out_vals[n] = {};
+    dt_helper_keys.retrieve_data(host_out_keys);
+    dt_helper_vals.retrieve_data(host_out_vals);
+   // sort_by_key with device policy guarantees stability, hence StableSort{} is passed
+    check_sort(host_out_keys, host_out_vals, n, StableSort{}, direction_tag);
+}
+
+template <typename StabilityTag, typename DirectionTag>
+void
+test_with_buffers(sycl::queue& q, StabilityTag stability_tag, DirectionTag direction_tag)
+{
+    constexpr int n = 1000000;
+    sycl::buffer<int> keys_buf{n};
+    sycl::buffer<int> vals_buf{n};
+
+    // generating data
+    {
+        sycl::host_accessor host_keys(keys_buf, sycl::write_only);
+        sycl::host_accessor host_vals(vals_buf, sycl::write_only);
+        generate_data(host_keys.begin(), host_vals.begin(), n, direction_tag);
+    }
+
+    // calling the algorithm
+    auto keys_begin = oneapi::dpl::begin(keys_buf);
+    auto vals_begin = oneapi::dpl::begin(vals_buf);
+    auto policy = TestUtils::make_device_policy<KernelName<class Buffer, StabilityTag, DirectionTag>>(q);
+    call_sort(policy, keys_begin, vals_begin, n, stability_tag, direction_tag);
+
+   // checking results
+   // sort_by_key with device policy guarantees stability, hence StableSort{} is passed
+   {
+        sycl::host_accessor host_keys(keys_buf, sycl::read_only);
+        sycl::host_accessor host_vals(vals_buf, sycl::read_only);
+        check_sort(host_keys.begin(), host_vals.begin(), n, StableSort{}, direction_tag);
+   }
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+#if TEST_DPCPP_BACKEND_PRESENT
+template <typename StabilityTag>
+void
+test_device_polcies(StabilityTag stability_tag)
+{
+    sycl::queue q = TestUtils::get_test_queue();
+    test_with_usm<sycl::usm::alloc::shared>(q, stability_tag, AscendingSort{});
+    test_with_usm<sycl::usm::alloc::device>(q, stability_tag, AscendingSort{});
+    test_with_buffers(q, stability_tag, DiscendingSort{});
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+template <typename StabilityTag, typename DirectionTag>
+void
+test_std_polcies(StabilityTag stability_tag, DirectionTag direction_tag)
+{
+    test_with_std_policy(oneapi::dpl::execution::seq, stability_tag, direction_tag);
+    test_with_std_policy(oneapi::dpl::execution::unseq, stability_tag, direction_tag);
+    test_with_std_policy(oneapi::dpl::execution::par, stability_tag, direction_tag);
+    test_with_std_policy(oneapi::dpl::execution::par_unseq, stability_tag, direction_tag);
+}
+
+template <typename StabilityTag>
+void
+test_all_policies(StabilityTag stability_tag)
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    test_device_polcies(stability_tag);
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+#if !TEST_DPCPP_BACKEND_PRESENT
```

Comment:
It looks as wrong test logic: if we have` dpcpp backend`, for example, we skip the tests with std policies....

Reply:
Now the standard policies are tested irrespective of device backend presence.